tests/test_connection.py

```diff
@@ -415,23 +431,40 @@ async def handler(request):
         assert header_key == b'x-test-header'
         assert header_value == b'My test header'
 
+def _trio_default_loose() -> bool:
+    assert re.match(r'^0\.\d\d\.', trio.__version__), "unexpected trio versioning scheme"
+    return int(trio.__version__[2:4]) < 25
 
 @fail_after(1)
-async def test_handshake_exception_before_accept():
+async def test_handshake_exception_before_accept() -> None:
     ''' In #107, a request handler that throws an exception before finishing the
     handshake causes the task to hang. The proper behavior is to raise an
     exception to the nursery as soon as possible. '''
     async def handler(request):
         raise ValueError()
 
-    with pytest.raises(ValueError):
+    with pytest.raises((BaseExceptionGroup, ValueError)) as exc:  # pylint: disable=possibly-used-before-assignment
         async with trio.open_nursery() as nursery:
             server = await nursery.start(serve_websocket, handler, HOST, 0,
                 None)
             async with open_websocket(HOST, server.port, RESOURCE,
                     use_ssl=False):
                 pass
 
+    if _trio_default_loose():
+        assert isinstance(exc.value, ValueError)
+    else:
+        # there's 4 levels of nurseries opened, leading to 4 nested groups:
+        # 1. this test
+        # 2. WebSocketServer.run
+        # 3. trio.serve_listeners
+        # 4. WebSocketServer._handle_connection
+        from trio.testing import RaisesGroup  # pylint: disable=import-outside-toplevel
```

Comment:
please move import to the module scope

Reply:
The import is only possible on trio>0.25, i.e. if `not _trio_default_loose()`. But I suppose I can call that on the module scope, or do a `try: ... except ImportError`